packages/expo-video/ios/VideoPlayer.swift

```diff
@@ -185,6 +213,10 @@ internal final class VideoPlayer: SharedRef<AVPlayer>, Hashable, VideoPlayerObse
     safeEmit(event: "sourceChange", arguments: newVideoPlayerItem?.videoSource, oldVideoPlayerItem?.videoSource)
   }
 
+  func onTimeUpdate(player: AVPlayer, timeUpdate: TimeUpdate) {
```

Comment:
I'm wondering if we need to pass the player to these lifecycle functions 🤔 In some of them they are unused and you can also just use `ref`/`pointer` property 🤔

Reply:
I agree, that In the case of the `VideoPlayer` class it doesn't make much sense to pass it, but other classes can observe multiple players and it comes in handy in that case.
For example, the `NowPlayingManager` also listens to some player events, and will receive events from all registered players, it also needs information on which player has emitted the event to function correctly.

I add the player to all of the events for future-proofing in case we ever need it.